### PR TITLE
Run app return graph

### PR DESCRIPTION
**Pull request by zhanglei1949**
## What do these changes do?

When a app runs on a simple graph and returns a `vertex data context`, we return a new graph.

## Related issue number

As a part of #1329 

Fixes

### CODE REVIEW

File: analytical_engine/core/grape_instance.h [hunk 1]
```diff
@@ -89,7 +89,14 @@ class GrapeInstance : public Subscriber {
   bl::result<void> unloadApp(const rpc::GSParams& params);
 
   bl::result<std::string> query(const rpc::GSParams& params,
-                                const rpc::QueryArgs& query_args);
+                                const rpc::QueryArgs& query_args,
+                                const std::string& out_context_key,
+                                std::shared_ptr<IFragmentWrapper>& wrapper);
+
+  bl::result<rpc::graph::GraphDefPb> merge_ctx_to_new_graph(
```
Comment (sighingnow) [inline analytical_engine/core/grape_instance.h:96]: Naming: camel case.

File: analytical_engine/core/fragment/arrow_projected_fragment.h [hunk 2]
```diff
@@ -399,6 +402,52 @@ class ArrowProjectedFragment
...
+        projected_frag_meta.GetKeyValue<label_id_t>("projected_v_label");
+    VLOG(1) << "projected label id: " << projected_label_id;
+
+    // First get out arrow array
+    std::vector<std::pair<std::string, std::string>> selector_list;
+    selector_list.emplace_back("result", "r");
+    std::string selector_string = generate_selectors(selector_list);
+    auto selector = Selector::ParseSelectors(selector_string).value();
+    BOOST_LEAF_AUTO(arrow_arrays,
+                    ctx_wrapper->ToArrowArrays(comm_spec, selector));
```
Comment (sighingnow) [inline analytical_engine/core/fragment/arrow_projected_fragment.h:423]: The `ToArrowArray` will do a copy.

The copy can be avoid, but it's fine for current implementation for simplification. Please add a `TODO` or `FIXME` here.

Comment (sighingnow) [reply analytical_engine/core/fragment/arrow_projected_fragment.h:423]: We could allocate the context on vineyard directly, but its fine to keep the current codepath. Anyway add a TODO.

File: analytical_engine/core/grape_instance.cc [hunk 3]
```diff
@@ -1097,8 +1121,24 @@ bl::result<std::shared_ptr<DispatchResult>> GrapeInstance::OnReceive(
...
-    BOOST_LEAF_AUTO(context_key, query(params, cmd->query_args));
-    r->set_data(context_key);
+    // For vertex context and arrowprojected fragment we return a new graph.
+    BOOST_LEAF_AUTO(graph_name, params.Get<std::string>(rpc::GRAPH_NAME));
+    BOOST_LEAF_AUTO(frag_wrapper,
+                    object_manager_.GetObject<IFragmentWrapper>(graph_name));
+    BOOST_LEAF_AUTO(return_graph, params.Get<bool>(rpc::RETURN_GRAPH));
+    std::string context_key = "ctx_" + generateId();
+    BOOST_LEAF_AUTO(context_result,
+                    query(params, cmd->query_args, context_key, frag_wrapper));
```
Comment (sighingnow) [inline analytical_engine/core/grape_instance.cc:1131]: Looks incorrect as the auto-assign `context_result` is never use.

Comment (sighingnow) [reply analytical_engine/core/grape_instance.cc:1131]: I suggest don't touch the implementation of `query()` and use the returned `context_key` instead.

File: analytical_engine/core/grape_instance.h (hunk 1 above)
Comment (zhanglei1949) [reply analytical_engine/core/grape_instance.h]: Resolved

**Pull request closed without merging**